internal/controllers/port/controller.go

```diff
@@ -51,9 +51,12 @@ var (
 	subnetDependency = dependency.NewDeletionGuardDependency[*orcv1alpha1.PortList, *orcv1alpha1.Subnet](
 		"spec.resource.addresses[].subnetRef",
 		func(port *orcv1alpha1.Port) []string {
-			subnets := make([]string, len(port.Spec.Resource.Addresses))
-			for i := range port.Spec.Resource.Addresses {
-				subnets[i] = string(port.Spec.Resource.Addresses[i].SubnetRef)
+			var subnets []string
+			if port.Spec.Resource != nil {
+				subnets = make([]string, len(port.Spec.Resource.Addresses))
+				for i := range port.Spec.Resource.Addresses {
+					subnets[i] = string(port.Spec.Resource.Addresses[i].SubnetRef)
+				}
```

Comment:
nit: It would be slightly more idiomatic to add an early return here rather than indenting the whole function. e.g.
```suggestion
			if port.Spec.Resource == nil {
				return nil
			}
			subnets := make([]string, len(port.Spec.Resource.Addresses))
			for i := range port.Spec.Resource.Addresses {
				subnets[i] = string(port.Spec.Resource.Addresses[i].SubnetRef)
			}
```
Same below.

Reply:
Thanks. Modified